specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2022-10-01-preview/machineLearningServices.json

```diff
@@ -1328,8 +1328,8 @@
         }
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/updateSchedules": {
-      "post": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/schedules": {
```

Comment:
>/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}/schedules [](http://example.com/codeflow?start=5&length=170)

This is adding a new resource type, which requires the full resource model: PUT, GET singleton, GET collection, DELETE. Also, the child type should not be embedded in the parent type. So it's a bigger change than you have here.

You could instead leave it as part of the computes type and support updates to the computeStartStop collection in the PUT of the computes type.

Reply:
This still adds a new resource type "workspaces/computes/schedules", which requires the full resource model (PUT, GET single, GET collection, DELETE, and [optional] PATCH, not just PATCH. If you want to do this with PATCH, you would implement it on the PATCH method of the parent resource type endpoint: "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.MachineLearningServices/workspaces/{workspaceName}/computes/{computeName}". The swagger already declares a PATCH method on that type, so you just need to implement it in the backend on PUT and/or PATCH requests to the parent resource type. Doing it that way means you don't need to add any new endpoints.